src/utils/pokemonItems.js

```diff
@@ -50,21 +50,21 @@ function PokemonItems() {
                 }));
             })
     }, [])
-
-    if (!pokemonNames) return null;
-
-    let pokemonRequests = []
-    pokemonNames.forEach(
-        (name) => {
-            pokemonRequests.push(GetPokemon(name))
-        }
-    )
     
-    Promise.all(pokemonRequests).then((allPokemonData) => {
-        setPokemonData(allPokemonData.map(pokemon => {
-            return Pokemon(pokemon)
-        }))
-    })
+    useEffect(() => {
+        let pokemonRequests = []
+        pokemonNames.forEach(
+            (name) => {
+                pokemonRequests.push(GetPokemon(name))
+            }
+        )
+
+        Promise.all(pokemonRequests).then((allPokemonData) => {
+            setPokemonData(allPokemonData.map(pokemon => {
+                return Pokemon(pokemon)
+            }))
+        })
+    }, [pokemonNames])
```

Comment:
I'm not sure, but I think this is the cause. Your components are keep updating this pokemonNames state, and it will run thru all the API calls even after they're done. 

So, by adding `[pokemonNames]`, it will tell the ReactHook, to just fetch the details whenever the pokemonNames changed.

Reply:
When you say I keep referring to the pokemonNames state, where is that? Do you mean each time I iterate in line 56, it is going to refetch all the pokemon names?

Reply:
Yo, bro. Because it keeps watching it, and do the same thing over and over again even if there's no change in the state. So, we added the `pokemonNames` into the useEffect telling it to do action only when the change is there.